include/up-cpp/communication/RpcClient.h

```diff
@@ -54,17 +54,57 @@ struct RpcClient {
 	                   std::optional<uint32_t> permission_level = {},
 	                   std::optional<std::string> token = {});
 
+	/// @brief A status code that can be returned in the response message's
+	///        attributes.
 	using Commstatus = v1::UCode;
 
+	/// @brief Represents the combined status possibilities for transport and
+	///        uProtocol network communications.
+	using Status = std::variant<v1::UStatus, Commstatus>;
+
 	/// @brief Contains either a UMessage (when successful) or a UStatus /
-	/// Commstatus when an error occurred. Commstatus would be set based
-	/// on the commstatus attributes field in any returned messages.
-	using MessageOrStatus =
-	    utils::Expected<v1::UMessage, std::variant<v1::UStatus, Commstatus>>;
+	///        Commstatus when an error occurred. Commstatus would be set based
+	///        on the commstatus attributes field in any returned messages.
+	using MessageOrStatus = utils::Expected<v1::UMessage, Status>;
+
+	/// @brief Callback connections for callbacks passed to invokeMethod()
+	using Connection = utils::callbacks::Connection<void, MessageOrStatus&&>;
```

Comment:
Yes. We have been asked by several C++ developers to provide a fully asynchronous interface that does not require them to implement a separate waiting thread for the futures. Since the native format for handling incoming messages on the transport is a callback, that option is provided in this interface.

It's worth noting that C++ futures do not work the same way as futures in Java. There are scenarios where they may not be acceptable.